driver/controller.go

```diff
@@ -133,10 +139,16 @@ func (d *Driver) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest)
 			},
 		},
 		VolumeContext: map[string]string{
-			PublishInfoVolumeName: volumeName,
+			PublishInfoVolumeName:  volumeName,
+			LuksEncryptedAttribute: luksEncrypted,
```

Comment:
With the included storage classes, there is one secret per volume. 

The `parameters` map in the `StorageClass`contains static key/value pairs but the CSI provisioner has support for limited templating (see [CSI Provisioner Parameters](https://kubernetes-csi.github.io/docs/Usage.html#csi-provisioner-parameters)) to pass along volume-specific information during the phases `ControllerPublish`, `NodeStage`, `NodeUnstage`

The included storage class use `${pvc.namespace}` and `${pvc.name}-luks-key` for `csiNodeStageSecretNamespace` and `csiNodeStageSecretName`. The CSI provisioner will evaluate those templates, load the referenced secret and pass it along in the publish context object. If there is no such secret, the volume will be stuck in the `Pending` phase and the CSI provisioner will emit a nice event for the volume saying that it can't find that secret.

So there is an explicit contract for using the included luks storage classes: If you use one of them to provision a volume, you must supply the volume's luks key in a secret.

The `LuksEncryptedAttribute` is static. We put the parameter `csi.cloudscale.ch/luks-encrypted: "true"` on a storage class to indicate that volumes that use this storage class are encrypted with luks.

Here's an example: Say you deploy an application with a PostgreSQL database and Elasticsearch. You would e.g. create persistent volume claims `postgres-pvc` and `elasticsearch-pvc` with `cloudscale-volume-ssd-luks` as a `StorageClass`. Since the contract of that storage class requires a luks key in a secret `${pvc.Name}-luks-key`, you have to deploy two secrets, `postgres-pvc-luks-key` and `elasticsearch-pvc-luks-key`. You can and should use different luks keys for each volume. 

Maybe that wasn't too clear in the `README.md`. After we merge this, we could work on improving the documentation for the CSI plugin.